dlk/python/dlk/templates/src/network.tpl.cpp

```diff
@@ -168,6 +168,32 @@ Network::~Network()
 #endif
 }
 
+int Network::memory() {
+  int total_mem_bytes = 0;
+  {% for node in graph.non_variables -%}
+  {% if node.available_buffer == '' -%}
+  {% for out_k in node.output_ops.keys() -%}
+  {% if node.output_ops.keys()|length > 1 -%}
+  total_mem_bytes += sizeof({{ node.dtype.cpptype() }}) * ({{ node.view.size_in_words_as_cpp }}); // {{ node.name + '_' + out_k }}_raw
+  {% else -%}
+  total_mem_bytes += sizeof({{ node.dtype.cpptype() }}) * ({{ node.view.size_in_words_as_cpp }}); // {{ node.name }}_raw
+  {%- endif %}
+  {%- endfor %}
+  {% elif node.available_buffer != '' and node.output_ops.keys()|length > 1 -%}
+  {% for out_k in node.output_ops.keys() -%}
+  {% if out_k != node.output_ops.keys()|list|first -%}
+  total_mem_bytes += sizeof({{ node.dtype.cpptype() }}) * ({{ node.view.size_in_words_as_cpp }}); // {{ node.name + '_' + out_k }}_raw
+  {%- endif %}
+  {%- endfor %}
+  {%- endif %}
+  {%- endfor %}
+#if !defined RUN_ON_FPGA
+  total_mem_bytes += sizeof(QUANTIZED_PACKED) * max_device_input_elems;
+  total_mem_bytes += sizeof(BIN_CONV_OUTPUT) * max_device_output_elems;
+#endif
+  return total_mem_bytes;
```

Comment:
There are several temporary buffer allocated by some operators (ex. kn2row_buf in `src/func/impl/generic/quantized_conv2d_kn2row.cpp`), but these are not counted yet.
I'm working to allocate these hidden temporary buffers in `Network::init()` ( #473 ), but it may take a while...
Until it is resolved, we should count these by hand...

Reply:
Ah thank you! Didn't know about these buffers.

Reply:
We can wait until they are allocated from `Network::init()`, if you want.
Or we can add it later and count by hand as you said :smile:

Reply:
These buffers are allocated by `std::make_unique`.

I found such buffers in following files:
- `src/func/conv2d.cpp`
- `src/func/*/batch_normalization.cpp`
- `src/func/impl/generic/quantized_conv2d_kn2row.cpp`
- `src/func/impl/arm_neon/quantized_conv2d_tiling.cpp`
- `src/func/impl/x86_avx/quantized_conv2d_tiling.cpp`
- `src/matrix/multiplication.cpp`

Reply:
Ah, yes, seems that for changing layout we need these temporaries...

Reply:
When #473 is fixed, buffers may be unified into single buffer, total buffer size is obvious.
It seems waiting for fixed is reasonable way.